src/lib.rs

```diff
@@ -179,21 +193,49 @@ impl Wiki {
         self.rate(article_id, RatingAction::Downvote)
     }
 
-    // Donate to an article
+    /*
+     * Donate to an article
+     * @TODO determine the better practice between explicitly setting `donation_amt`
+     * as a contract method argument, and implicitly deriving it from `env::attached_deposit()`
+     */
+    #[payable]
+    pub fn donate(&mut self, article_id: u64) {
+
+        self.panic_on_nonexistent_article(article_id);
+
+        let donation_amt: u128 = env::attached_deposit();
+
+        if donation_amt.lt(&MIN_DONATION_AMOUNT) {
+            env::panic(format!("Donation amount cannot be less than {:?}", MIN_DONATION_AMOUNT).as_bytes());
+        }
+        if donation_amt.gt(&MAX_DONATION_AMOUNT) {
+            env::panic(format!("Donation amount cannot be greater than {:?}", MAX_DONATION_AMOUNT).as_bytes());
+        }
+
+        let meta = self.meta.get(&article_id).unwrap();
+        let author_id = meta.author;
+
+        // @TODO define the business logic where donation amount is split among the author and contributors
+        let account_id: AccountId = author_id.parse().unwrap();
+        Promise::new(account_id).transfer(donation_amt);
```

Comment:
I remember reading somewhere that you need to return the final Promise for this to work properly; by using the return keyword, or remove the last semicolon.

Another question is how we can transfer to multiple accounts?

Reply:
According to [this writeup](https://www.near-sdk.io/zero-to-hero/beginner/actions#adding-transfer), it's okay either we return the Promise, or not :) The Transfer action would be carried out either way.

Also we have the unit test `donate_successfully_to_article` to verify the transaction did happen. I'd say we're good to go.

Reply:
https://near-sdk.io//promises/token-tx

In this doc, returning Promise will let the explorer and client to verify if the transfer is successful or not. Not returning Promise will return successfully regardless of transfer status.

Reply:
That's a very good read, thanks :) Will fix.